src/UnicycleGenerator.cpp

```diff
@@ -641,4 +694,15 @@ std::shared_ptr<CoMHeightTrajectoryGenerator> UnicycleGenerator::addCoMHeightTra
     return m_pimpl->comHeightGenerator;
 }
 
+std::shared_ptr<DCMTrajectoryGenerator> UnicycleGenerator::addDCMTrajectoryGenerator()
```

Comment:
As far as I know one returns a `shared_ptr` if one wants to have the ownership attached to the pointer, but I think that here the ownership should belong only to the `UnicycleGenerator` object. So I suggest to make `m_pimpl->dcmTrajectoryGenerator` an `unique_ptr` and return the raw pointer `m_pimpl->dcmTrajectoryGenerator.get()`  
What do you think?

Reply:
I prefer not to return the raw pointer, otherwise the user class will have to check the existence of that generator every time it tries to access it. In the way you propose, all the generators would be deleted as soon as the ``UnicycleGenerator`` will go out of scope. Instead, by using a shared pointer, the generators will still be accessible without the need to check the pointer, but it will not be possible to generate new trajectories.